### PR TITLE
[BUBO-72] My rewards page

### PR DESCRIPTION
**Pull request by icedevera**
[[BUBO-72](https://vacuum.atlassian.net/browse/BUBO-72?atlOrigin=eyJpIjoiZTU4NmNhNTRjMDAyNDkyMzlmODVjMDVhMGZhMWM2MDgiLCJwIjoiaiJ9)] 

image

[BUBO-72]: https://vacuum.atlassian.net/browse/BUBO-72?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by matejfalat — COMMENTED** [review]
Haven't tested it, just went through the code. Noted a few minor things but otherwise looking good 👍

File: src/app/(findings)/my-findings/rewards/page.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  prefetchGetMyFindingsRewardsSize,
+} from '@/lib/queries/reward/getMyFindingsRewards'
+
+const myFindingsRewardsPage = async () => {
+  const {user} = await requirePageSession()
+
+  await Promise.all([
+    prefetchGetMyFindingsRewards(user.id, {
+      limit: MY_FINDINGS_REWARDS_PAGE_SIZE,
+    }),
```
[the remark this answers is not in the document]
Comment (icedevera) [reply src/app/(findings)/my-findings/rewards/page.tsx]: Yeah, you're right, thanks for this

File: src/components/sections/finding/rewards/MyFindingsRewardsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import TableHeadWithSort from '@/components/ui/TableHeadWithSort'
+import {MyFindingsRewardsSorting} from '@/lib/types/enums'
+
+export const MY_FINDINGS_REWARDS_PAGE_SIZE = 7
+
+const MyFindingsRewardsTable = () => {
+  const [page] = useSearchParamsNumericState('page', 1)
+  const [sortParams, setSortParams] = useSortingSearchParams(
+    MyFindingsRewardsSorting,
+  )
```
[the remark this answers is not in the document]
Comment (icedevera) [reply src/components/sections/finding/rewards/MyFindingsRewardsTable.tsx]: Good call. Seemed like the page search param was not updating when the sort parameters changed. Modified the `useSortingSearchParams` and `TableHeaderWithSort` to fix this 👍

**Review by matejfalat — APPROVED** [review]
LGTM

**Pull request merged**